### PR TITLE
Dcaplin/dev

### PR DESCRIPTION
**Pull request by davidcaplin**

### CODE REVIEW

File: core-accounting-fe/src/core-accounting/resource/ReferenceManyLines.tsx [hunk 1]
```diff
@@ -1,21 +1,42 @@
...
     >
-      <Datagrid bulkActionButtons={false}>
-        <TextField source="code" sortable={false} />
-        <MoneyField source="amount" />
-        <MoneyField source="balance" />
-        {children}
-        <TimeAgoField label="Created" source="createdAt" />
-      </Datagrid>
+      {/* A list within a reference field doesnt normally work (it ignores the target, but you can reset the target w/ filter, and filter for values where the scope is equal to the current record id (user.id)) */}
+      <List exporter={false} filter={{ scope: record.id }} filters={lineFilters} >
```
Comment (maxschridde1494) [inline core-accounting-fe/src/core-accounting/resource/ReferenceManyLines.tsx:28]: i.e. lineFilters is the name of the optional prop that is passed into the ReferenceManyLines component

File: core-accounting-fe/src/core-accounting/resource/index.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  static List = Table
+  static ReferenceManyLines = ReferenceManyLines
+  static Show = LineShow
+  // // static Show = ShowSession
+  // static Icon = PlaylistAddCheckIcon
+}
+
+
+
+export class Transfers {
```
Comment (maxschridde1494) [inline core-accounting-fe/src/core-accounting/resource/index.ts:17]: what is this?

File: core-accounting-fe/src/core-accounting/resource/table.tsx [hunk 3]
```diff
@@ -22,4 +35,29 @@ export const Table: React.FC<{datagridProps?: any; tableProps?: any, children?:
       </Datagrid>
     </List>
   )
-}
\ No newline at end of file
+}
+
+
+export const TransferList = () => (
```
Comment (maxschridde1494) [inline core-accounting-fe/src/core-accounting/resource/table.tsx:41]: what is this?

File: core-accounting-fe/src/core-accounting/resource/table.tsx [hunk 4]
```diff
@@ -1,17 +1,30 @@
...
 
 import { MoneyField } from './MoneyField'
 import { TimeAgoField } from './TimeAgoField'
 
-export const Table: React.FC<{datagridProps?: any; tableProps?: any, children?: any}> = ({datagridProps, tableProps, children}) => {
+import { PolymorphicReferenceField } from './PolymorphicReferenceField'
+import { Card, CardContent } from '@mui/material'
+import MailIcon from '@mui/icons-material/MailOutline';
+
+export const PostFilterSidebar = () => (
```
Comment (maxschridde1494) [inline core-accounting-fe/src/core-accounting/resource/table.tsx:12]: again, this is application specific. We can add the ability for filter / aside / sort params to be passed into the table component (similar to what we are doing with the children), but they shouldn't live within the core-accounting-fe pkg

File: core-versions-fe/src/core-versions/mockVersions.ts [hunk 5]
```diff
@@ -1,14 +1,63 @@
-export async function mockVersions({count = 15}): Promise<any> {
```
Comment (maxschridde1494) [inline core-versions-fe/src/core-versions/mockVersions.ts:1]: what is the difference between the mockVersions.ts and mock-versions.ts files? can one of them be deleted?

File: test-core-fe/src/data/mock-data-provider.ts [hunk 6]
```diff
@@ -32,15 +33,114 @@ const comments = (await mockComments({count: 50})).map((comment) => {
...
+//a transfer is 2 lines, so 8 mock transfers returns 16 lines
+const lines = (await mockTransfers({ count: 150 })).map((line, idx) => {
+  console.log("asdf")
+  const user = faker.helpers.arrayElement(users)
+  const resource = faker.helpers.arrayElement([
+    faker.helpers.arrayElement(users),
+    faker.helpers.arrayElement(companies),
+  ])
+  let scope, partnerScope
+  if (!(idx % 2)) { //line 1
```
Comment (maxschridde1494) [inline test-core-fe/src/data/mock-data-provider.ts:66]: this logic won't create lines with matching scope & partner scope. you are regenerating a new user and resource with each pass through the line array (not by pair of lines).

File: core-versions-fe/src/core-versions/mockVersions.ts (hunk 5 above)
Comment (davidcaplin) [reply core-versions-fe/src/core-versions/mockVersions.ts:1]: Yeah, the mock-versions.ts is an old file that I had renamed mockVersions.ts and then edited, not sure why it's still in there, mock-can be deleted

File: core-accounting-fe/src/core-accounting/mockLines.ts [hunk 7]
```diff
@@ -7,44 +7,144 @@ declare global {
...
+if (Math.random() > .5) {
+  code = 'buy_aqd'
+  partnerCode = 'spend_aqd'
+}
+else { //line 2
+  code = 'spend_aqd'
+  partnerCode = 'buy_aqd'
+}
+
+export async function mockTransfers({ count = 8 }): Promise<any> {
```
Comment (maxschridde1494) [inline core-accounting-fe/src/core-accounting/mockLines.ts:32]: rename to mockLines

Comment (maxschridde1494) [reply core-accounting-fe/src/core-accounting/mockLines.ts:32]: also add ```transfer_codes = ['buy_tokens', 'spend_tokens']``` arg to mockLines

File: core-accounting-fe/src/core-accounting/mockLines.ts [hunk 8]
```diff
@@ -7,44 +7,144 @@ declare global {
...
+// }
+// for each pair of credit / debit lines
+//   grab a user & a company
+//   line 1 has user as account / scope and company as partner account / scope
+//   line 2 has company as account / scope and user as partner account / scope
 
-export async function mockLines({count = 15}): Promise<any> {
+//assuming they can both buy and spend from each other
+//what about transfers?
+let code, partnerCode
```
Comment (maxschridde1494) [inline core-accounting-fe/src/core-accounting/mockLines.ts:22]: delete.

File: core-accounting-fe/src/core-accounting/mockLines.ts (hunk 7 above)
Comment (maxschridde1494) [reply core-accounting-fe/src/core-accounting/mockLines.ts:32]: first line of pair gets code = transfer_codes[0] and second line of pair gets code = transfer_codes[1]

File: core-accounting-fe/src/core-accounting/resource/index.ts (hunk 2 above)
Comment (maxschridde1494) [reply core-accounting-fe/src/core-accounting/resource/index.ts:17]: please remove.

File: core-accounting-fe/src/core-accounting/resource/table.tsx (hunk 4 above)
Comment (maxschridde1494) [reply core-accounting-fe/src/core-accounting/resource/table.tsx:12]: please remove

File: core-accounting-fe/src/core-accounting/resource/table.tsx (hunk 3 above)
Comment (maxschridde1494) [reply core-accounting-fe/src/core-accounting/resource/table.tsx:41]: please remove

File: core-comments-fe/src/core-comments/resources/comments/TableList.tsx [hunk 9]
```diff
@@ -31,7 +31,10 @@ const commentFilters = [
   <DateInput source="created_at_gte" label="After" />,
   <ReferenceInput alwaysOn label="Author" source="author_id" reference="User" perPage={10}>
```
Comment (maxschridde1494) [inline core-comments-fe/src/core-comments/resources/comments/TableList.tsx:32]: can't assume the reference is a User. This is a polymorphic ref

File: core-versions-fe/src/core-versions/mockVersions.ts (hunk 5 above)
Comment (maxschridde1494) [reply core-versions-fe/src/core-versions/mockVersions.ts:1]: please remove mock-versions.ts if redundant.

File: core-comments-fe/src/core-comments/resources/comments/TableList.tsx (hunk 9 above)
Comment (davidcaplin) [reply core-comments-fe/src/core-comments/resources/comments/TableList.tsx:32]: WIP

File: core-accounting-fe/src/core-accounting/index.ts [hunk 10]
```diff
@@ -1,7 +1,11 @@
 import { Lines } from './resource'
+import { ReferenceManyLines } from './resource/ReferenceManyLines'
 
 export { mockLines } from './mockLines'
 
 export class CoreAccounting {
   static Lines = Lines
+  static Resource = Lines
+  static ReferenceManyLines = ReferenceManyLines
+  // static Resource = Transfers
```
Comment (maxschridde1494) [inline core-accounting-fe/src/core-accounting/index.ts:10]: please remove this comment.

File: core-comments-fe/src/core-comments/resources/comments/PolymorphicReferenceField.tsx [hunk 11]
```diff
@@ -15,7 +15,7 @@ export const PolymorphicReferenceField = ({
   source,
 }: PolymorphicReferenceFieldProps) => {
   const record = useRecordContext()
-
+  console.log("PL ", record )
```
Comment (maxschridde1494) [inline core-comments-fe/src/core-comments/resources/comments/PolymorphicReferenceField.tsx:18]: please remove

File: core-accounting-fe/src/core-accounting/mockLines.ts [hunk 12]
```diff
@@ -7,44 +7,84 @@ declare global {
 Array.prototype.sample = function () {
   return this[Math.floor(Math.random() * this.length)]
 }
-
-export async function mockLines({count = 15}): Promise<any> {
+//assuming they can both buy and spend from each other
+//what about transfers?
+export async function mockLines({ count = 8 }): Promise<any> {
```
Comment (maxschridde1494) [inline core-accounting-fe/src/core-accounting/mockLines.ts:12]: code array should be passed in as prop

File: core-accounting-fe/src/core-accounting/mockLines.ts [hunk 13]
```diff
@@ -7,44 +7,84 @@ declare global {
...
-      scope: faker.datatype.uuid(),
-      code: ['payment', 'transfer', 'spend'].sample(),
-      amount: {
-        amount: faker.commerce.price(1, 1000, 0),
-        formatted_amount: faker.commerce.price(1, 1000, 0, '$'),
-        currency: {
-          code: 'EQD',
-          symbol: '$',
+    let lines = []
+    let code = ['buy_aqd','spend_aqd'].sample()
```
Comment (maxschridde1494) [inline core-accounting-fe/src/core-accounting/mockLines.ts:16]: sample from the code arg

File: test-core-fe/src/data/mock-data-provider.ts [hunk 14]
```diff
@@ -32,15 +33,116 @@ const comments = (await mockComments({count: 50})).map((comment) => {
...
+    resource_type: resource.type,
+
+    // author: user,
+    // resource: resource
+  }
+})
+
+//a transfer is 2 lines, so 8 mock transfers returns 16 lines
+let user, resource;
+const lines = (await mockLines({ count: 100 })).map((line, idx) => {
```
Comment (maxschridde1494) [inline test-core-fe/src/data/mock-data-provider.ts:59]: should take accounts & codes as arg

**Pull request merged**